File: C/Program16.c
/////////////////////////////////////////////////////////////////////////
//                                                                     //
//  File name :     Program16.c                                        //
//  Description :   Used to Display Factors of given number            //
//  Author :        Pallavi Omprakash Malewar                          //
//  Date :          19/07/2025                                         //
//                                                                     //
/////////////////////////////////////////////////////////////////////////


#include<stdio.h>

void DisplayFactorsNonFactors(int iNo)
{
    int iCount = 0;

    if(iNo < 0)
    {
        iNo = -iNo;     // Updator 
    }
    for(iCount = 1 ; iCount <= (iNo / 2) ; iCount++)
    {
        if((iNo % iCount) == 0)
        {
            printf("Factor is : %d\n" , iCount);
        }
        else
        {
            printf("Non Factor is : %d\n" , iCount);
        }
    }
    
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number to display factors & non factors : ");
    scanf("%d" , &iValue);

    DisplayFactorsNonFactors(iValue);
    
    return 0;
}

// Time Complexity O(N/2)

File: JAVA/Program16.java
//////////////////////////////////////////////////////////////////////////////////////////
//                                                                                      //
//  File name :     Program16.java                                                      //
//  Description :   Used to Count Digits in Given Number                                //
//  Author :        Pallavi Omprakash Malewar                                           //
//  Date :          25/08/2025                                                          //
//                                                                                      //
//////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Digits
{
    public int countDigits(int iNo)
    {
        int iCount = 0;

        iNo = Math.abs(iNo);        // Inbuilt method for Filter(To handle negative Numbers)

        if(iNo == 0)
        {
            return 1;
        }

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        while(iNo > 0)
        {
            iCount++;
            iNo = iNo / 10;
        }

        return iCount;
    }
}

class Program16
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iRet = 0;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        Digits dobj = new Digits();

        iRet = dobj.countDigits(iValue);

        System.out.println("Total Digits in "+ iValue + " : " + iRet);

        sobj.close(); 
    }
}